connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/CustomApacheHttpClient.java

```diff
@@ -97,14 +99,22 @@ private static PoolingHttpClientConnectionManager createConnectionManager() {
   @Override
   public HttpCommonResult execute(
       HttpCommonRequest request, @Nullable ExecutionEnvironment executionEnvironment) {
-    var apacheRequest = ApacheRequestFactory.get().createHttpRequest(request);
     try {
+      var apacheRequest = ApacheRequestFactory.get().createHttpRequest(request);
+      ProxyHandler proxyHandler =
+          new ProxyHandler(apacheRequest.getScheme(), apacheRequest.getUri().getHost());
```

Comment:
Looks like we are creating a new proxy handler for every new request. I think this should be done once.